diff --git a/scp.c b/scp.c
index 1faa9a5..f05efb0 100644
--- a/scp.c
+++ b/scp.c
@@ -153,7 +153,7 @@ int throughlocal = 1;
 int sshport = -1;
 
 /* This is the program to execute for the secure connection. ("ssh" or -S) */
-char *ssh_program = _PATH_SSH_PROGRAM;
+char *ssh_program;
 
 /* This is used to store the pid of ssh_program */
 pid_t do_cmd_pid = -1;
@@ -459,6 +459,8 @@ main(int argc, char **argv)
 	char *sftp_direct = NULL;
 	long long llv;
 
+  ssh_program = _PATH_SSH_PROGRAM;
+
 	/* Ensure that fds 0, 1 and 2 are open or directed to /dev/null */
 	sanitise_stdfd();
 
